lti_consumer/models.py

```diff
@@ -538,6 +550,42 @@ def get_lti_consumer(self):
 
         return self._get_lti_1p1_consumer()
 
+    def get_lti_1p3_redirect_uris(self):
+        """
+        Return pre-registered redirect uris or sensible defaults
+        """
+        if self.config_store == self.CONFIG_EXTERNAL:
+            # TODO: Add support for CONFIG_EXTERNAL for LTI 1.3.
+            raise NotImplementedError
+
+        if self.config_store == self.CONFIG_ON_DB:
+            redirect_uris = self.lti_1p3_redirect_uris
+            launch_url = self.lti_1p3_launch_url
+            deep_link_launch_url = self.lti_advantage_deep_linking_launch_url,
+        else:
+            redirect_uris = self.block.lti_1p3_redirect_uris
+            launch_url = self.block.lti_1p3_launch_url
+            deep_link_launch_url = self.block.lti_advantage_deep_linking_launch_url,
+
+        return self._get_lti_1p3_redirect_uris_with_defaults(
+            redirect_uris,
+            launch_url,
+            deep_link_launch_url
+        )
+
+    def _get_lti_1p3_redirect_uris_with_defaults(self, redirect_uris, launch_url, deep_link_url):
+        """
+        Return provided redirect_uris if set, else use launch/deep_link as defaults
```

Comment:
Is there any harm in always including the launch URL and deep link URL if one exists, since they are already there as trusted URLs?

Reply:
Do we want to have defaults? The exchange of registered redirect URIs is a required part of the registration of a tool, right? Why are we making it an optional part of the configuration? Is it your experience that tools don't reliably and explicitly provide this information during registration?

Reply:
Yes that’s exactly what I do now! if the redirect Uris are not specifically set then it uses the launch url and deep linking url as the defaults. Once you set anything in the redirect Uris field though, you would then need to include all possible redirect Uris. 

I did it this way so we it won’t affect the current setup of previous deployments.

The general use case we’re supporting here is that if the launch url and target link uri are the same then they only need to set the target link uri and/or deep linking uri. But there are tools where the redirect uri (the place we return the token to) is not the same as the target link uri (the final place the tool will redirect to once it redirects to the tools redirect uri). 

The valid redirect Uris should technically only ever be set to places the token should be returned to. No need to include the target link uri in this list because it wouldn’t know what to do with a token anyway. 

Hope that makes sense. Lots of redirect Uris and terminology here!

Reply:
@MichaelRoytman I only implemented the defaults as a measure to account for what would otherwise be a breaking change for existing deployments. I'm fine with removing them as long as everyone understands that after upgrading to this version, their integration would cease to function until they added the appropriate urls to the new redirect uri's field.